api/src/main/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.pmiops.workbench.survey;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import org.pmiops.workbench.access.AccessTierService;
+import org.pmiops.workbench.db.dao.UserAccessTierDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbUserAccessTier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NewUserSatisfactionSurveyServiceImpl implements NewUserSatisfactionSurveyService {
+
+  private final UserAccessTierDao userAccessTierDao;
+  private AccessTierService accessTierService;
+  private final Clock clock;
+
+  @Autowired
+  public NewUserSatisfactionSurveyServiceImpl(
+      UserAccessTierDao userAccessTierDao, AccessTierService accessTierService, Clock clock) {
+    this.userAccessTierDao = userAccessTierDao;
+    this.accessTierService = accessTierService;
+    this.clock = clock;
+  }
+
+  @Override
+  public boolean eligibleToTakeSurvey(DbUser user) {
+    if (user.getNewUserSatisfactionSurvey() != null) {
+      return false;
+    }
+
+    DbAccessTier registeredAccessTier = this.accessTierService.getRegisteredTierOrThrow();
```

Comment:
style nit: `this.` can generally be omitted because the variable name is unambiguous.  The exception to this rule is in constructors.

Reply:
Thanks, I thought it was required. Updated, along with a few other instances.